main.nf

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env nextflow
+
+
+// check parameters
+if (!params.inputDir) {
+    exit 1, "inputDir parameter is missing."
+}
+if (!params.outputDir) {
+    exit 1, "outputDir parameter is missing."
+}
+if ( !new File(params.inputDir).exists()) {
+    exit 1, "The input folder does not exists."+params.inputDir+"\n"
+}
+if (!params.outPrefix) {
+    exit 1, "Output prefix parameter is missing."
+}
+if (!params.debug) {
+    exit 1, "Debug prefix parameter is missing."
+}
+if (!params.localOutDir) {
+    params.localOutDir='bam'
+}
+
+pairFiles_ch = Channel.fromFilePairs( params.inputDir+"/*{1,2}.fastq.gz", size: 2, checkIfExists: true )
+
+process ERValign {
+    // tag ${sample}
+    
+    // executor configuration
+    time '8h'
+    // memory '35.GB'
+    scratch true
+    storeDir params.outputDir
+
+    // other configuration
+    echo true
+    errorStrategy 'terminate'
+
+    input:
+    val(outPrefix) from params.outPrefix
+    val(localOutDir) from params.localOutDir
+    val(cpus) from params.cpus
+    val(limitMemory) from params.limitMemory
+    val(debug) from params.debug
+    tuple val(sample), file(reads) from pairFiles_ch
+    
+    output:
+    path ( "${localOutDir}/${outPrefix}Aligned.sortedByCoord.out.bam" ) into bam_ch
+    path ( "${localOutDir}/${outPrefix}Aligned.sortedByCoord.out.bam.bai" ) into bai_ch
+
+    shell:
+    template 'ERValign.sh'
+}
+
+process ERVcount {
+//     tag ${sample}
+
+    // executor configuration
+    time '3h'
+    memory '8.GB'
+    scratch true
+    storeDir params.outputDir
+
+    // other configuration
+    echo true
+    errorStrategy 'terminate'
+    mode = 'BED'
+    stageInMode 'symlink'
+    
+    input:
+    val(outPrefix) from params.outPrefix
+    val(debug) from params.debug
+    path bam from bam_ch
+    path bai from bai_ch
+    
+    output: 
+    path (params.outPrefix+"ERVresults.txt") into final_results_ch
```

Comment:
I used the second version that you mention. Am I missing something?

Reply:
I'm the one missing something in my comment. The second version should have read as 
params.outPrefix+'ERVresults.txt'

My point is that here you can avoid the double quotes because there is no variable to evaluate. It's a detail that repeated tens of times in the workflow makes it slower.

Reply:
Got it!